### ISSUE TITLE
AWS .NET SDK: Presigned Urls with ResponseHeaderOverrides are not working

### ISSUE THREAD

**Issue by KarlKl** [localstack/localstack#3522]
# Type of request:

[x] bug report
[ ] feature request

# Detailed description

Creating a presigned URL with ResponseHeaderOverrides with the AWS .NET SDK, results in an invalid signature.

AWSSDK.S3 3.5.3.9

## Expected behavior

Generated presigned Url should have a valid signature.

## Actual behavior

Error is thrown when the URL is requested (Browser/Postman/C#):
403 response content: `<Error><Code>SignatureDoesNotMatch</Code><Message>The request signature we calculated does not match the signature you provided. Check your key and signing method.</Message>...`

# Steps to reproduce

## Command used to start LocalStack

```
# create bucket and put test file in it, nothing special here
awslocal s3api create-bucket --bucket local-bucket
echo "hello" > test.txt
awslocal s3 cp test.txt s3://local-bucket/test.txt
```

## Client code (AWS .NET SDK)
```
var config = new AmazonS3Config
{
    ServiceURL = "localhost:4566",
    UseHttp = true,
    ForcePathStyle = true
};
var s3Client = new AmazonS3Client("test", "test", config);
var request = new GetPreSignedUrlRequest
{
    BucketName = "local-bucket",
    Key = "test.txt",
    Expires = DateTime.Now.AddDays(1),
    Verb = HttpVerb.GET,
    ResponseHeaderOverrides = new ResponseHeaderOverrides
    {
        ContentType = "text/plain" // when I comment this line out, everything is working great
    }
};
var presignedUrl = s3Client.GetPreSignedURL(request);
// GET the url eg in the browser
```

**Comment by KarlKl** [comment]
Possibly related to #3266?

**Comment by niemyjski** [comment]
I ran into this today as well. This commit is what introduced this issue (https://github.com/localstack/localstack/commit/6d463be76ad416e0418645c9e8a0212dfdefec3a) #3092 . The latest amazon sdk uses the v4 signature method for `us-east-1`. However when using s3 an internal service map overrides this value and sets it to 2.

**Comment by ejsmith** [comment]
Same issue here.

**Comment by niemyjski** [comment]
My workaround is to use the `localstack/localstack:0.12.4` image

**Comment by KarlKl** [comment]
Thanks @niemyjski.
But this is not a suitable workaround for us as we are dependent on a bug fix made in 0.12.5 for the .NET SDK #2165. 🙈

**Comment by niemyjski** [comment]
@KarlKl then what can we do to resolve this? I've already stepped through this with the latest amazon .NET SDK and I don't see any work around.

**Comment by KarlKl** [comment]
@niemyjski I have tried it with v0.12.4 and **can not confirm** that this was working back then, I get the same error. Is this even possible? 🤔 

I think the signature creation is somehow not correctly implemented, because this works (but should not):
- create presigned url **without override**
- **add** the url **parameter**(eg `content-type=test`) to the presigned url
- the error **does not appear** and content-type is `test`, although the url has been changed and signature should not be valid anymore

Don't know if this is a bug in localstack or in botocore, or if the behaviour has been changed in v4. 🤔

**Comment by jcunhafonte** [comment]
> My workaround is to use the `localstack/localstack:0.12.4` image

@niemyjski This doesn't solve my issue. Can you give more details, please?

**Comment by niemyjski** [comment]
I don't have any more information to add in addition to my previous comments. 0.12.4 fixed my issue as it reverted to the release before the new code was introduced. You'd need to dig in and debug it to see what the remaining issues are.

**Comment by whummer** [comment]
Thanks for reporting @ejsmith @KarlKl @niemyjski @jcunhafonte . This should (hopefully) be fixed in #3603 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!

**Comment by KarlKl** [comment]
I can confirm that it is working in most cases. 🙈 The provided code example originally added to this report is working. 👍 
Some strange behavior I have found is the following:
```
...
ResponseHeaderOverrides = new ResponseHeaderOverrides
{
    ContentType = "text/plain",
    ContentDisposition = "attachment;filename=test.txt",  // does work
    ContentDisposition = "attachment; filename=test.txt" // does NOT work
                                     ^ space char breaks signature
}
...
```
The space char "breaks" the signature, the signature created isn't valid and I'm getting a 403 like before. I think the original AWS implementation does somehow parse the Content-Disposition and trims spaces before signature creation. 🤔 

This is of course easy to work around. I don't know how much effort this would be to fix, or if you prefer to leave it as known bug. @monty16597?
